BSP/sharpMemoryLCD/LS013B7DH03.c

```diff
@@ -0,0 +1,206 @@
+#include "LS013B7DH03.h"
+
+#include <stdbool.h>
+#include <stm32l4xx_hal.h>
+#include "fonts.h"
+
+/** Update pixel data */
+#define CMD_DATA_UPDATE (0x01)
+/** Clear internal memory */
+#define CMD_CLEAR       (0x04)
+
+#define SCREEN_HEIGHT   (128)
+#define SCREEN_LENGTH   (128)
+
+SPI_HandleTypeDef SpiHandle;
+
+/**
+ * @brief set SS pin
+ *
+ * @param ss slave select true to select
+ */
+static void LCDslaveSelect(bool ss)
+{
+    if(ss)
+    {
+        HAL_GPIO_WritePin(GPIOD, GPIO_PIN_0, GPIO_PIN_SET);
+    }
+    else
+    {
+        HAL_GPIO_WritePin(GPIOD, GPIO_PIN_0, GPIO_PIN_RESET);
+    }
+}
+
+/**
+ * @brief set DISP pin
+ *
+ * @param en enable true to desplay info from memory
+ */
+static void LCDdisplayEnable(bool en)
+{
+    if(en)
+    {
+        HAL_GPIO_WritePin(GPIOD, GPIO_PIN_6, GPIO_PIN_SET);
+    }
+    else
+    {
+        HAL_GPIO_WritePin(GPIOD, GPIO_PIN_6, GPIO_PIN_RESET);
+    }
+}
+
+
+void sharpMemoryLCD_init(void)
+{
+    __HAL_RCC_GPIOB_CLK_ENABLE();
+    __HAL_RCC_GPIOD_CLK_ENABLE();
+    __HAL_RCC_TIM3_CLK_ENABLE();
+
+    GPIO_InitTypeDef GPIO_InitStruct;
+    GPIO_InitStruct.Mode = GPIO_MODE_AF_PP;
+    GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_LOW;
+
+    GPIO_InitStruct.Pull = GPIO_NOPULL;
+    GPIO_InitStruct.Alternate = GPIO_AF2_TIM3;
+    GPIO_InitStruct.Pin = GPIO_PIN_4;
+    /* Pin B0 is EXTCOMIN driven by TIM3_CH1 */
+    HAL_GPIO_Init(GPIOB, &GPIO_InitStruct);
+
+    GPIO_InitStruct.Pin = GPIO_PIN_1;
+    /* Pin D1 is SPI2 CLK */
+    HAL_GPIO_Init(GPIOD, &GPIO_InitStruct);
+
+    GPIO_InitStruct.Pin = GPIO_PIN_4;
+    /* Pin D4 is SPI2 MOSI */
+    HAL_GPIO_Init(GPIOD, &GPIO_InitStruct);
+
+    GPIO_InitStruct.Mode = GPIO_MODE_OUTPUT_PP;
+    GPIO_InitStruct.Pin = GPIO_PIN_0;
+    /* Pin D0 is SPI2 NSS */
+    /* It is not driven by SPI2 peripheral because need to be enabled with high level */
+    HAL_GPIO_Init(GPIOD, &GPIO_InitStruct);
+    LCDslaveSelect(false);
+
+    GPIO_InitStruct.Pin = GPIO_PIN_6;
+    /* Pin D6 is DISP (for enabling display) */
+    HAL_GPIO_Init(GPIOD, &GPIO_InitStruct);
+    LCDdisplayEnable(false);
+
+
+    /* Output a 60kHz signal to EXTCOMIN pin */
+    TIM_HandleTypeDef TimHandle;
+    TimHandle.Instance = TIM3;
+    TimHandle.Init.Period        = 65535;
+    TimHandle.Init.Prescaler     = 60000; // counter is counting at 120M/60k => 2kHz
+    TimHandle.Init.ClockDivision = TIM_CLOCKDIVISION_DIV4; // 500Hz
+    TimHandle.Init.CounterMode   = TIM_COUNTERMODE_UP;
+    TimHandle.Init.Period        = 8; //500/8 => 60Hz period
+
+    if (HAL_TIM_PWM_Init(&TimHandle) != HAL_OK)
+    {
+        assert_param(false);
+    }
```

Comment:
It is odd, but what do think of 
```suggestion
    assert_param(HAL_TIM_PWM_Init(&TimHandle) == HAL_OK);
```
Or introducing a initStatus var?
I Find it weird to to assert_param(false) directly.

Reply:
your proposal is  good. Anyway in the meantime I though about this and I think it's better to return a status to caller.
These functions should never fail. It's more to have a stop point during debug. I will change it

Reply:
Approved, a return status is the best way to go. "It will never fail".